src/Compilers/Core/Portable/Syntax/SyntaxListBuilderExtensions.cs

```diff
@@ -32,7 +34,7 @@ public static SyntaxList<TNode> ToList<TNode>(this SyntaxListBuilder builder)
                 return new SyntaxList<TNode>();
             }
 
-            return new SyntaxList<TNode>(builder.ToListNode().CreateRed());
+            return new SyntaxList<TNode>(builder.ToListNode()!.CreateRed());
         }
 
         public static SeparatedSyntaxList<TNode> ToSeparatedList<TNode>(this SyntaxListBuilder builder) where TNode : SyntaxNode
```

Comment:
💡 The implementation allows `builder` to be null. I accounted for this in my suggested refactoring.